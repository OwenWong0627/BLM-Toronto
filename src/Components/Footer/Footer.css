.footer-container {
    background-color: #edeeee;
    padding: 0rem 0 2rem 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  
  /* Social Icons */
  .social-icon-link {
    color: #000;
    font-size: 24px;
  }
  
  .social-media {
    max-width: 1000px;
    width: 100%;
  }
  
  .social-media-wrap {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 90%;
    max-width: 1000px;
    margin: 20px auto 0 auto;
  }
  
  .social-icons {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    width: 240px;
    margin-top: -10px;
  }
  
  .social-logo {
    color: #000;
    justify-self: center;
    cursor: pointer;
    text-decoration: none;
    font-size: 2rem;
    display: flex;
    align-items: center;
  }

  .footer-logo-image {
    margin-left: 0.5rem;
    width: 1.8rem;  
   }
  
  .website-rights {
    color: #000;
  }
  
  @media screen and (max-width: 820px) {
  
    .social-media-wrap {
      flex-direction: column;
    }
  }
  
  @media screen and (max-width: 768px) {
  }