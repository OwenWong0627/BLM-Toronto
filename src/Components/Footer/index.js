import React from 'react';
import './Footer.css';
import { Link } from 'react-router-dom';

/**
 * This component represents the footer in the web app, it has a logo which will redirect users to the home page
 * This component has a Github icon that links the web app's Github repository
 */
function Footer() {
   return (
      <div className='footer-container'>
         <div className='social-media-wrap'>
            <Link data-testid='footer-logo' to="/" className="social-logo">
               BLM-Toronto <img className="footer-logo-image" src='/BLM-Toronto-Logo.svg' alt='Footer Logo' />
            </Link>
         </div>
         <div className='social-media-wrap'>
            <a
               className="social-logo"
               data-testid='github-link'
               target='_blank'
               aria-label='Github'
               href='https://github.com/OwenWong0627/BLM-Toronto'
               rel='noopener noreferrer'
            >
               <img data-testid="github-icon" src='/github-logo.svg' className='github-icon' alt='Github Logo' />
            </a>
         </div>
         <div className='social-media-wrap'>
            <small data-testid='website-rights' className='website-rights'>Â© Owen Wong & Leo Wang. All rights reserved</small>
         </div>
      </div>
   );
}

export default Footer;