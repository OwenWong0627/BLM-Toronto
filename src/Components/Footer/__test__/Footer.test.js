import React from 'react';
import ReactDOM from 'react-dom';
import Footer from '..';

import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it("renders without crashing", () => {
   const div = document.createElement("div");
   ReactDOM.render(<Router><Footer /></Router>, div);
});

it('Renders the correct content', () => {
   const { getByTestId } = render(<Router><Footer /></Router>);
   expect(getByTestId("footer-logo")).toHaveTextContent("BLM-Toronto");
   expect(getByTestId("footer-logo")).toHaveAttribute("href", "/");
   expect(getByTestId("footer-logo")).toContainHTML("img");
   expect(getByTestId("github-link")).toHaveAttribute("href", "https://github.com/OwenWong0627/BLM-Toronto");
   expect(getByTestId("github-link")).toHaveAttribute("target", "_blank");
   expect(getByTestId("github-link")).toContainElement(getByTestId("github-icon"));
   expect(getByTestId("website-rights")).toHaveTextContent("Â© Owen Wong & Leo Wang. All rights reserved");
});

it('matches snapshot', () => {
   const tree = renderer.create(<Router><Footer /></Router>).toJSON();
   expect(tree).toMatchSnapshot();
});