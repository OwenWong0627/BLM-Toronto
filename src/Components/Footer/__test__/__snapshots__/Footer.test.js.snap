// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="footer-container"
>
  <div
    className="social-media-wrap"
  >
    <a
      className="social-logo"
      data-testid="footer-logo"
      href="/"
      onClick={[Function]}
    >
      BLM-Toronto 
      <img
        alt="Footer Logo"
        className="footer-logo-image"
        src="/BLM-Toronto-Logo.svg"
      />
    </a>
  </div>
  <div
    className="social-media-wrap"
  >
    <a
      aria-label="Github"
      className="social-logo"
      data-testid="github-link"
      href="https://github.com/OwenWong0627/BLM-Toronto"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Github Logo"
        className="github-icon"
        data-testid="github-icon"
        src="/github-logo.svg"
      />
    </a>
  </div>
  <div
    className="social-media-wrap"
  >
    <small
      className="website-rights"
      data-testid="website-rights"
    >
      Â© Owen Wong & Leo Wang. All rights reserved
    </small>
  </div>
</div>
`;
